src/biokbase/narrative/exception_util.py

```diff
@@ -40,23 +57,28 @@ def transform_job_exception(e, error=None):
     ServerError - thrown by the server, usually due to a 500 (ish) exception
     HTTPError - a more mundane HTTP exception
 
-    error is an optional message that will be passed to the resultant
-    NarrativeException
+    error is an optional message that will be passed to the resultant NarrativeException
     """
+    service_not_found_error_codes = [404, 502, 503]
+    service_timeout_error_codes = [504, 598, 599]
+    service_internal_error_codes = [500]
+
     if isinstance(e, EEServerError):
         return NarrativeException(e.code, e.message, e.name, "ee2", error)
+
     if isinstance(e, HTTPError):
         code = e.response.status_code
-        if code == 404 or code == 502 or code == 503:
+        if code in service_not_found_error_codes:
             # service not found
             msg = "A KBase service is currently unavailable."
-        elif code == 504 or code == 598 or code == 599:
+        elif code in service_timeout_error_codes:
             # service timeout
             msg = "There was a temporary network connection error."
-        elif code == 500:
+        elif code in service_internal_error_codes:
             # internal error. dunno what to do.
             msg = "An internal error occurred in the KBase service."
         else:
             msg = "An untracked error occurred."
         return NarrativeException(e.response.status_code, msg, "HTTPError", "network", error)
-    return NarrativeException(-1, str(e), "Exception", "unknown", error)
+
+    return NarrativeException(-1, str(e), type(e).__name__, "unknown", error)
```

Comment:
altered the output of the fallback so that it captures the type of the exception thrown, rather than just calling it a generic "Exception"